src/ItemList.js

```diff
@@ -1,25 +1,82 @@
 import React, { useState, useEffect } from 'react';
 import Item from './Item';
 import { useHistory } from 'react-router-dom';
+import filter from './lib/filter';
+import { differenceInDays, fromUnixTime } from 'date-fns';
 
 function ItemList(props) {
   const [query, setQuery] = useState('');
-  const [queryArray, setQueryArray] = useState([]);
+  const [queryObj, setQueryObj] = useState({
+    week: [],
+    month: [],
+    longer: [],
+    inactive: [],
+  });
   let history = useHistory();
   const redirect = () => {
     history.push('/additems');
   };
 
   useEffect(() => {
-    // [filter-list] 2. Comparison function to filter shopping list and create a search results array
-    // [filter-list] (Remove accidental space character from query, make lowercase, and prevent regex errors)
-    const resultsArray = props.list.filter((itemObj) => {
-      return itemObj['itemName']
-        .toLowerCase()
-        .replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
-        .includes(query.toLowerCase().trim());
+    const resultsObj = {
+      week: [],
+      month: [],
+      longer: [],
+      inactive: [],
+    };
+
+    props.list.forEach((item) => {
+      if (item.purchaseDates.length < 1 || !(item.purchaseEstimates?.length > 0)) {
+        resultsObj['inactive'].push(item);
+      } else {
+        const lastPurchase = fromUnixTime(
+          item.purchaseDates[item.purchaseDates.length - 1].seconds,
+        );
+        const lastInterval = differenceInDays(Date.now(), lastPurchase);
+        const lastEstimate =
+          item.purchaseEstimates[item.purchaseEstimates.length - 1];
+        const daysRemaining = lastEstimate - lastInterval;
+        item.daysRemaining = daysRemaining;
+
+        if (lastInterval >= 2 * lastEstimate) {
+          resultsObj['inactive'].push(item);
+        } else if (daysRemaining <= 7) {
+          resultsObj['week'].push(item);
+        } else if (daysRemaining <= 30) {
+          resultsObj['month'].push(item);
+        } else {
+          resultsObj['longer'].push(item);
+        }
+      }
```

Comment:
Because this whole `useEffect` is pretty long, it could use some comments to divide up the sections, I think. Might be nice to think of chunks of it that could be extracted out into their own functions.  idk